src/rich-log/domain.hh

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <clean-core/macros.hh>
+
+namespace rlog
+{
+namespace verbosity
+{
+enum type : int
+{
+    /// TRACE is used to provide diagnostic information verbosely and in very small steps
+    /// e.g. logging in each iteration of an algorithm
+    /// per default, trace should be compile-time DISABLED (as it can and will slow down the app)
+    Trace = 0,
+
+    /// DEBUG is used to provide diagnostic information that usually do not matter
+    /// e.g. logging the arguments to an algorithm
+    /// per default, debug should be compile-time ENABLED but runtime DISABLED
+    /// this means all message can be viewed post-mortem
+    Debug = 1,
+
+    /// INFO is used to log interesting information that USERS of your algorithm might want to know about
+    /// e.g. the selected graphics card, filenames for large save operations, UI actions
+    /// per default, this logging is enabled
+    Info = 2,
+
+    /// WARNING is used to log suspicious calls / inputs
+    /// e.g. loading empty files, showing empty meshes, degenerate faces
+    /// while they are not strictly errors, you should minimize false positives (users will start to ignore them)
+    /// per default, this logging is enabled
+    Warning = 3,
+
+    /// ERRROR is used for clear errors and wrong use of APIs
+    /// e.g. trying to load a nonexistant file, providing unsupported argument combinations
+    /// these errors do not crash the program, but are clearly wrong
+    /// per default, this logging is enabled, will break, and will record stacktraces
```

Comment:
Personally don't like it when LOG(Error, "...") causes a debug break since many things are "errors" but not necessarily programmer-relevant defects. In the same vein i wouldn't record a stacktrace here

Reply:
hm yeah good point. FATAL breaks and records then per default

Reply:
but I think it's valueable to have a good guideline on the enum values here, otherwise it's wild west